src/backends/pyplot.jl

```diff
@@ -1113,7 +1113,13 @@ function _before_layout_calcs(plt::Plot{PyPlotBackend})
             end
             pyaxis."label"."set_fontsize"(py_thickness_scale(plt, axis[:guidefontsize]))
             pyaxis."label"."set_family"(axis[:guidefontfamily])
-            pyaxis."label"."set_rotation"(axis[:guidefontrotation])
+
+            if (letter == :y && !RecipesPipeline.is3d(sp))
```

Comment:
How about adding
```julia
 || (letter == :z && RecipesPipeline.is3d(sp))
```
here? Or would add `-90` to the rotation in that case?

Reply:
Isn't having letter z equivalent of being 3d plot?

For 3d plots no rotations need to be specified

Reply:
But the z label in the 3D Plots you posted should be rotated analogously to y labels in 2D Plots, no?

Reply:
... to save whitespace for longer labels

Reply:
In v0.29.9, before the guidefontrotation was ever introduced, this is behavior:
```
 p = plot(1:10, 1:10, 1:10, ylabel="ylab", xlabel="xlab",zlabel="zlab")
pyplot()
gr()
ploltly()
```
![Figure_1](https://user-images.githubusercontent.com/24591123/79289401-178dff00-7f04-11ea-8152-5c66c8903611.png)

![gr](https://user-images.githubusercontent.com/24591123/79289570-971bce00-7f04-11ea-9b12-b53c381be66d.png)
![newplot](https://user-images.githubusercontent.com/24591123/79289571-98e59180-7f04-11ea-8290-a8cf5d51d8dc.png)


That is in case of 3d plot, the guides are not rotated by default (in gr() and pyplot()). There is already some inconsistence among backends. So I kept what was default for pyplot back in 0.29.9 (to avoid breaking)